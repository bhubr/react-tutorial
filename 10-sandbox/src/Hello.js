import React from 'react'

const Hello = props => <h2>{props.name}</h2>

export default Hello
