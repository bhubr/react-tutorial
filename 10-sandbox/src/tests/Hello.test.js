// Link.react.test.js
import React from 'react';
import Hello from '../Hello';
import renderer from 'react-test-renderer';

test('Hello world', () => {
  const component = renderer.create(
    <Hello name="Dr. Robert" />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // tree.props.onMouseEnter();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();

  // // manually trigger the callback
  // tree.props.onMouseLeave();
  // // re-rendering
  // tree = component.toJSON();
  // expect(tree).toMatchSnapshot();
});
