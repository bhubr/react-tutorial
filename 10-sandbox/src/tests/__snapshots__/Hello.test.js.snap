// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello world 1`] = `
<h2>
  Dr. Robert
</h2>
`;
